### PR TITLE
Zq/update group norm by aclnn

### PR DESCRIPTION
**Pull request by NeosZhang**
## Motivation and Context

update group norm by aclnn

## Description

update group norm by aclnn
image

## Checklist
**Before PR**:

- [ ] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/ascend/functions/group_norm.cpp [hunk 1]
```diff
@@ -4,31 +4,66 @@
...
+                                    diopiConstTensorHandle_t mean, diopiConstTensorHandle_t rstd, int64_t numGroups) {
+    AscendTensor inputAt(input);
+    AscendTensor gradWeightAt(gradWeight);
+
+    if (!inputAt.defined()) {
+        return diopiSuccess;
+    }
+
+    if (inputAt.numel() == 0) {
+        diopiScalar_t zeroScalar = constructDiopiScalarT(diopi_dtype_float64, 0.0);
```
Comment (yangbofun) [inline impl/ascend/functions/group_norm.cpp:42]: 直接调用aclnninplacezero

File: impl/ascend/aclnn/adaptor.hpp [hunk 2]
```diff
@@ -108,6 +109,17 @@ inline aclScalar* createAclScalarFromDiopiScalar(const diopiScalar_t* scalar) {
 
 inline aclIntArray* createAclIntArrayFromDiopiSize(const diopiSize_t size) { return ::aclCreateIntArray(size.data, size.len); }
 
+template <size_t N = 0>
```
Comment (yangbofun) [inline impl/ascend/aclnn/adaptor.hpp:112]: 把N=0删了，不加默认值。

File: impl/ascend/functions/group_norm.cpp (hunk 1 above)
Comment (NeosZhang) [reply impl/ascend/functions/group_norm.cpp]: done

File: impl/ascend/aclnn/adaptor.hpp (hunk 2 above)
Comment (NeosZhang) [reply impl/ascend/aclnn/adaptor.hpp]: done

**Review by zhangzefeng92 — APPROVED** [review]

File: impl/ascend/functions/group_norm.cpp [hunk 3]
```diff
@@ -4,31 +4,65 @@
...
+                                    diopiConstTensorHandle_t gradOutput, diopiConstTensorHandle_t input, diopiConstTensorHandle_t weight,
+                                    diopiConstTensorHandle_t mean, diopiConstTensorHandle_t rstd, int64_t numGroups) {
+    AscendTensor inputAt(input);
+    AscendTensor gradWeightAt(gradWeight);
+
+    if (!inputAt.defined()) {
+        return diopiSuccess;
+    }
+
+    if (inputAt.numel() == 0) {
```
Comment (yangbofun) [inline impl/ascend/functions/group_norm.cpp:41]: 这里input已经不存在数据了，但是还对bias和weight求导。没有意义。看看torch的行为是什么样的。

Comment (NeosZhang) [reply impl/ascend/functions/group_norm.cpp:41]: torch的行为和代码逻辑相同
对于NCHW的输入，当N为0时，
image

对于NCHW的输入，当H为0时，
image

File: impl/ascend/functions/group_norm.cpp [hunk 4]
```diff
@@ -4,31 +4,65 @@
...
-        .addOutput(out)
-        .addOutput(saveMean)
-        .addOutput(saveInvstd)
-        .run();
+    int64_t n = inputAt.shape(0);
+    int64_t c = inputAt.shape(1);
+    int64_t hw = inputAt.numel() / (n * c);
+    eps = (eps < 1e-5) ? 1e-5 : eps;
+
+    DIOPI_ASCEND_CALL_ACLNN(aclnnGroupNorm, ctx, input, weight, bias, n, c, hw, numGroups, eps, out, saveMean, saveInvstd);
```
Comment (NeosZhang) [inline impl/ascend/functions/group_norm.cpp:27]: eps为负值的测例需要跳过

**Review by yangbofun — APPROVED** [review]

**Pull request merged**